### ISSUE TITLE
survive nick changes

### ISSUE THREAD

**Issue by fauno** [mmb/weechat-otr#123]
if any end of the conversation changes nickname, the session is dropped and authentication status is lost.  wouldn't it be better to identify people by something more stable, like usernames?  maybe use the host mask if available (not an ip address i mean).

**Comment by mmb** [comment]
Nicks are registered with a nickserv and authenticated on most networks. Usernames are arbitrary strings that the client can set to whatever it wants. Is there a use case where users frequently change nicks?

**Comment by fauno** [comment]
"Matthew M. Boedicker" <notifications@github.com> writes:

> Nicks are registered with a nickserv and authenticated on most
> networks. Usernames are arbitrary strings that the client can set to
> whatever it wants. Is there a use case where users frequently change
> nicks?

in my experience, no one sets the username (i just leave it empty on
weechat, no idea what pidgin decides to use as username) but you can
change your nick anytime for fun and profit.  most people i talk with
don't even bother to register them...  and you can always end with an
alternative nick if your connection isn't stable.

**Comment by tribut** [comment]
Nicks are potentially long-lived (if properly registered) and unique, while usernames are neither unique nor long-lived (because hardly anybody cares, so if someone switches clients they won't bother changing it for example).
Nick change detection would be an interesting feature (for example using the full user@host while the conversation is active), but otherwise we should really stick with nicks.

**Comment by Mikaela** [comment]
```
2015-11-26 09:14:00+0200 < Mikaela> #123: WeeChat supports account-notify which makes the IRCd tell it when someone identifies or unidentifies. It also performs /who to everyone on all channels on join to het the accounts of people who joined the channel before.  Would it help here?
2015-11-26 09:14:32+0200  * Mikaela cannot GitHub at the moment
2015-11-26 09:17:32+0200  * nils_2 can github
2015-11-26 09:35:11+0200 < Mikaela> Feel free,  I am going to take a shower
```

Thanks a lot for GitHubbing @weechatter !

**Comment by mmb** [comment]
I agree that nicks are still the best choice for the primary identifier. Automatic nick change detection may open up new attack vectors.
